astropy/io/fits/file.py

```diff
@@ -30,6 +30,9 @@
 if HAS_BZ2:
     import bz2
 
+from astropy.utils.compat.optional_deps import HAS_FSSPEC
+if HAS_FSSPEC:
+    import fsspec
```

Comment:
I agree!  I didn't make it a local import because I didn't encounter local imports elsewhere in Astropy, and I believe flake8 told me off when I tried, but I think it is the right choice here.